File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        left = 0
        right = len(heights) - 1
        maxarea = 0

        while left < right:
            lower = min(heights[left], heights[right])
            maxarea = max(maxarea, abs(left - right) * lower)
            if lower == heights[left]:
                left += 1
            elif lower == heights[right]:
                right -= 1
        
        return maxarea
        

File: Data Structures & Algorithms/is-subsequence/submission-0.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int p = 0;
        for (const auto& c : t)
        {
            if (c == s[p])
            {
                p++;
                if (p == s.length())
                {
                    break;
                }
            }
        }
        return p == s.length();
    }
};